### PR TITLE
feat: adds common instructions, checks, verbose

### PR DESCRIPTION
**Pull request by R-K-H**
This improves the script for initializing proposals by:
- including common instructions
- improving feedback to the user
- adding more logging
- checks for balances
- sleep before trigger

### CODE REVIEW

File: scripts/initializeProposal.ts [hunk 1]
```diff
@@ -15,37 +16,170 @@ const PANTERA_PUBKEY = new PublicKey(
   "BtNPTBX1XkFCwazDJ6ZkK3hcUsomm1RPcfmtUrP6wd2K"
 );
 
-async function main() {
-  const senderAcc = await token.getOrCreateAssociatedTokenAccount(
+const COST_DEPLOY = 4 * LAMPORTS_PER_SOL
+
+// Transfer
+const transferToAccount = async (
```
Comment (metaproph3t) [inline scripts/initializeProposal.ts:22]: this is to build a transfer instruction, right? would suggest `buildTreasuryTransferInstruction` or something similar

File: scripts/initializeProposal.ts [hunk 2]
```diff
@@ -15,37 +16,170 @@ const PANTERA_PUBKEY = new PublicKey(
...
-    senderAcc.address,
-    receiverAcc.address,
+    originAcc.address,
+    destinationAcc.address,
+    daoTreasury,
+    transferAmount
+  );
+
+  console.log('Transfer instructions');
+  console.log(transferIx);
```
Comment (metaproph3t) [inline scripts/initializeProposal.ts:73]: no need to log this IMO

File: scripts/initializeProposal.ts [hunk 3]
```diff
@@ -15,37 +16,170 @@ const PANTERA_PUBKEY = new PublicKey(
...
+
+  return transferIx;
+}
+
+// Memo
+const createMemo = async (memoText: string) => {
```
Comment (metaproph3t) [inline scripts/initializeProposal.ts:79]: same thing on naming

File: scripts/initializeProposal.ts [hunk 4]
```diff
@@ -15,37 +16,170 @@ const PANTERA_PUBKEY = new PublicKey(
...
+  console.log('Memo text');
+  console.log(memoText);
+  console.log('Memo text length');
+  console.log(memoText.length);
+  const byteLengthOfMemo = Buffer.byteLength(memoText)
+  if (byteLengthOfMemo >= 566) {
```
Comment (metaproph3t) [inline scripts/initializeProposal.ts:85]: Where'd you get this from?

File: scripts/initializeProposal.ts [hunk 5]
```diff
@@ -15,37 +16,170 @@ const PANTERA_PUBKEY = new PublicKey(
...
+  );
+
+  console.log('Burn instructions');
+  console.log(burnIx);
+
+  return burnIx;
+}
+
+// Migrate
+const migrateProgram = async () => {}
```
Comment (metaproph3t) [inline scripts/initializeProposal.ts:144]: if we don't have the logic, we shouldn't include

File: scripts/initializeProposal.ts [hunk 6]
```diff
@@ -15,37 +16,170 @@ const PANTERA_PUBKEY = new PublicKey(
...
+    data: proposalIx.data,
   };
 
-  await initializeProposal(ix, "https://hackmd.io/@0xNallok/Hy2WJ46op");
+  console.log('Proposal instruction');
+  console.log(ix);
+
+  // Sleep for review
+  console.log('Sleeping for 60s, press ctrl + c to cancel');
+  await new Promise(f => setTimeout(f, 60000));
```
Comment (metaproph3t) [inline scripts/initializeProposal.ts:180]: smort

File: scripts/initializeProposal.ts [hunk 7]
```diff
@@ -15,37 +16,170 @@ const PANTERA_PUBKEY = new PublicKey(
...
+
+  await initializeProposal(ix, "https://google.com");
```
Comment (metaproph3t) [inline scripts/initializeProposal.ts:182]: should we also log this?

**Comment by R-K-H** [comment]
Closed in favor of https://github.com/metaDAOproject/futarchy/pull/100

**Pull request closed without merging**